client/babel.config.js

```diff
@@ -0,0 +1,8 @@
+module.exports = {
+  presets: [
+    '@vue/cli-plugin-babel/preset',
+  ],
+  plugins: [
+    '@babel/plugin-proposal-optional-chaining',
+  ],
+};
```

Comment:
So there's this thing called ECMAScript -- its basically a standard agreed upon by the tech overlords discussing what a general purpose 'scripting' language should be able to do. Every now and then, the standard changes. Javascript is based on that standard and changes with it. The issue is that some browsers don't support the 'new' ECMAScript standard and can't run the 'new' Javascript that came with it, so code that uses that standard might not be compatible with every browser.

Babel is a 'transpiler' that basically takes the 'new' javascript and converts into the 'old' javascript so it can run anywhere.

This config file is basically just saying we're using babel's standard (preset) transpiler that automatically detects the platform it will be running on and does all the conversions for it. That's opposed to the developer having more fine grain control over what conversions happen for what browsers. The plugin added is 'optional chaining' which Vue 2.0 (the version we're using) doesn't support out of box because its based on older ECMAScript standard than the one that includes that feature. 

Optional chaining basically gives us the freedom to test whether a layer of an object hierarchy even exists before going deeper. As an example, with the following

grandparent?.parent?.child?

we test whether grandparent exists before looking at parent and we test whether parent exists before looking at child. That makes it really easy to work with objects that don't necessarily have every layer present because you make it impossible to check for member variables of a NULL variable. It's used in a couple of places where we want to verify something in a nested hierarchy has been initialized before we allow a certain operation.

Reply:
And yes, we can add some documentation to the file that explains this at a high level!

Reply:
Love the explanation, thanks!